File: Problems/longest common prefix.c
char* longestCommonPrefix(char** strs, int strsSize) {
    char *str = malloc(1000 * sizeof(char));
    int j = 0, k = 0;

    while (strs[0][j] != '\0') {
        char ch = strs[0][j];
        for (int i = 1; i < strsSize; i++) {
            if (strs[i][j] == '\0' || strs[i][j] != ch) {  
                str[k] = '\0';
                return str;
            }
        }
        str[k++] = ch;
        j++;
    }
    str[k] = '\0';  
    return str;
}

File: Day 8/longest common prefix.java
class Solution {
    public String longestCommonPrefix(String[] strs) {
       StringBuilder str1 = new StringBuilder();
       for(int i=0;i<strs[0].length();i++){
        char ch = strs[0].charAt(i);
        for(int j=1;j<strs.length;j++){
            if(i>=strs[j].length() || strs[j].charAt(i)!=ch) 
                return str1.toString();
        }
        str1.append(ch);
       }
       return str1.toString();
    }
}